azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -1091,20 +1098,88 @@ public int getExecutionId() {
+      metaData.put("submitUser", flow.getSubmitUser());
+      metaData.put("executionId", String.valueOf(flow.getExecutionId()));
+      metaData.put("startTime", String.valueOf(flow.getStartTime()));
+      metaData.put("submitTime", String.valueOf(flow.getSubmitTime()));
+      return metaData;
+    }
+
+    @Override
+    public synchronized void handleEvent(final Event event) {
+      if (FlowRunner.this.azkabanEventReporter == null) {
+        FlowRunner.this.logger.debug("Event reporter is not enabled");
+        return;
+      }
+
+      if (event.getType() == Type.FLOW_STARTED) {
+        final FlowRunner flowRunner = (FlowRunner) event.getRunner();
+        final ExecutableFlow flow = flowRunner.getExecutableFlow();
+        FlowRunner.this.logger.info("Flow started: " + flow.getId());
+        FlowRunner.this.azkabanEventReporter.report(event, getFlowMetadata(flowRunner));
+      } else if (event.getType() == Type.FLOW_FINISHED) {
+        final FlowRunner flowRunner = (FlowRunner) event.getRunner();
+        final ExecutableFlow flow = flowRunner.getExecutableFlow();
+        FlowRunner.this.logger.info("Flow ended: " + flow.getId());
+        final Map<String, String> flowMetadata = getFlowMetadata(flowRunner);
+        flowMetadata.put("endTime", String.valueOf(flow.getEndTime()));
+        FlowRunner.this.azkabanEventReporter.report(event, flowMetadata);
+      }
+    }
+  }
+
   private class JobRunnerEventListener implements EventListener {
 
     public JobRunnerEventListener() {
     }
 
+    private synchronized Map<String, String> getJobMetadata(final JobRunner jobRunner) {
+      final ExecutableNode node = jobRunner.getNode();
+      final Map<String, String> metaData = new HashMap<>();
+      metaData.put("jobId", node.getId());
+      metaData.put("executionID", String.valueOf(node.getExecutableFlow().getExecutionId()));
+      metaData.put("flowName", node.getExecutableFlow().getId());
+      metaData.put("startTime", String.valueOf(node.getStartTime()));
+      metaData.put("jobType", String.valueOf(node.getType()));
+      metaData.put("jobProxyUser",
+          jobRunner.getProps().getString("user.to.proxy", null));
+      return metaData;
+    }
+
     @Override
     public synchronized void handleEvent(final Event event) {
-
       if (event.getType() == Type.JOB_STATUS_CHANGED) {
         updateFlow();
       } else if (event.getType() == Type.JOB_FINISHED) {
-        final JobRunner runner = (JobRunner) event.getRunner();
-        final ExecutableNode node = runner.getNode();
         final EventData eventData = event.getData();
+        final JobRunner jobRunner = (JobRunner) event.getRunner();
+        final ExecutableNode node = jobRunner.getNode();
+        if (FlowRunner.this.azkabanEventReporter != null) {
+          final Map<String, String> jobMetadata = getJobMetadata(jobRunner);
+          // data only available post call to prepareJob method, and hence added only to the
```

Comment:
Do you mean azakbanHost? 
I thought it is defined in the config. You should be able to access it anywhere via Guice.

Reply:
Yes, I did mean azkabanHost. Though it's possible to access the hostname via something like `AzkabanExecutorServer.getApp().getHost()`, I found it difficult to test the same. The getApp method is static and mockito doesn't allow mocking of static methods. I did not want to introduce another library for this. Let me know if there's a better way.

Reply:
AzkabanExecutorServer.getApp() is deprecated. Don't use in the new code.
Try azkaban.ServiceProvider instead for now to get this information.

Reply:
done.